pkg/cache/cache.go

```diff
@@ -21,8 +28,8 @@ type CacheItem[T any] struct {
 	expiresAt int64
 }
 
-func NewCache[T any](options CacheOptions) (*Cache[T], error) {
-	c := &Cache[T]{
+func NewBasicCache[T any](options CacheOptions) (*BasicCache[T], error) {
```

Comment:
I've made these changes (and a couple of others) in https://github.com/bacalhau-project/bacalhau/pull/2448 (after cherry-picking from this PR) so that I can keep this PR focused on non-cache changes, and have sent a review request if you have time.